### PR TITLE
interface 

### PR DESCRIPTION
**Pull request by Hendzior**
Added interfece for reading data in csv and xml.
Organize code.

### CODE REVIEW

File: src/main/java/XmlPersonalDataAccess.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          int totalPersons = listOfpersons.getLength();
+
+
+          for (int s = 0; s < totalPersons; s++) {
+
+              person = new Person();
+              Node firstPersonNode = listOfpersons.item(s);
+              Element firstPersonElement = (Element)firstPersonNode;
+
+              NodeList firstNameList = firstPersonElement.getElementsByTagName("first-name");
```
Comment (myslinskipawel) [inline src/main/java/XmlPersonalDataAccess.java]: Extract to method ```readXmlTagValue("first-name")``` and use in others below using different tag name as parameter. Same code reused! AWESOME :)

File: src/main/java/XmlPersonalDataAccess.java [hunk 2]
```diff
@@ -9,85 +11,133 @@
...
+            DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+            Document document = documentBuilder.parse(file);
+            document.getDocumentElement().normalize();
+            NodeList listOfpersons = document.getElementsByTagName("person");
+            int totalPersons = listOfpersons.getLength();
+
+            for (int s = 0; s < totalPersons; s++) {
+
+
+                Node firstPersonNode = listOfpersons.item(s);
```
Comment (myslinskipawel) [inline src/main/java/XmlPersonalDataAccess.java]: It's not 'first'. It's every one person from list. 'personNode' name would be better
Same below

File: src/main/java/XmlPersonalDataAccess.java [hunk 3]
```diff
@@ -9,85 +11,133 @@
...
+            logger.error("** Parsing error" + ", line "
+                    + err.getLineNumber() + ", uri " + err.getSystemId());
+
+
+        } catch (SAXException e) {
+            logger.error("error while reading", e);
+        } catch (Throwable t) {
+            logger.error("error while reading", t);
+        }
+        logger.info("file {} read succesfully", file);
```
Comment (myslinskipawel) [inline src/main/java/XmlPersonalDataAccess.java]: Message 'read successful' will be processed even if thread went to catch clause (exception was thrown). How can we do this better?

File: src/main/java/XmlPersonalDataAccess.java [hunk 4]
```diff
@@ -9,85 +11,133 @@
...
+        } catch (ParserConfigurationException pce) {
+            logger.error("error writing file {}. " + file + "Message: {}", pce.getMessage());
 
-          }
+        } catch (TransformerException tfe) {
+            logger.error("error writing file {}. " + file + "Message: {}", tfe.getMessage());
+
+        }
+
+        logger.info("file {} write succesfully", file);
```
Comment (myslinskipawel) [inline src/main/java/XmlPersonalDataAccess.java]: Should not be logged off there was an exception

File: mylogs.log [hunk 5]
```diff
@@ -1645,3 +1645,854 @@
 19-01-2019 12:28:08 INFO  App:58 - Number of females: 1
```
[the remark this answers is not in the document]
Comment (Hendzior) [reply mylogs.log]: That make sense, thanks :)

File: src/main/java/XmlPersonalDataAccess.java (hunk 3 above)
Comment (myslinskipawel) [reply src/main/java/XmlPersonalDataAccess.java]: yup 👍

**Pull request merged**